### PR TITLE
Miscellaneous changes (update events)

### PR DESCRIPTION
**Pull request by nevi-me**
* Makes margin::events public
* Remove unused variables

### CODE REVIEW

**Comment by eyevz** [comment]
Looks sensible to me.

File: programs/margin-pool/src/events.rs [hunk 1]
```diff
@@ -12,6 +12,8 @@ pub struct PoolCreated {
     pub authority: Pubkey,
     pub payer: Pubkey,
     pub summary: MarginPoolSummary,
+    pub version: u8,
+    pub pool_bump: u8,
```
Comment (suscd) [inline programs/margin-pool/src/events.rs:16]: why include this?

File: programs/margin/src/events.rs [hunk 2]
```diff
@@ -56,6 +56,7 @@ pub struct PositionBalanceUpdated {
 
 #[event]
 pub struct PositionTouched {
+    pub margin_account: Pubkey,
```
Comment (suscd) [inline programs/margin/src/events.rs:59]: why is this necessary? The margin account is already recorded in the invoke event, including it here also for every single position seems annoyingly redundant (given log size is already an issue).

Comment (eyevz) [reply programs/margin/src/events.rs:59]: No objection. There are certainly savings to be had if we're willing to grab `Pubkey` fields from the instruction calls.

File: programs/margin/src/instructions/register_position.rs [hunk 3]
```diff
@@ -82,7 +82,6 @@ pub fn register_position_handler(ctx: Context<RegisterPosition>) -> Result<()> {
     let position = account.get_position_by_key(&key).require()?;
 
     emit!(events::PositionRegistered {
-        margin_account: ctx.accounts.margin_account.key(),
```
Comment (suscd) [inline programs/margin/src/instructions/register_position.rs:85]: A potential issue with removing this is you don't know the margin account for the case where users directly register a position (as opposed to through invoke, which records an extra event that mentions the margin account).

**Review by suscd — APPROVED** [review]
